airbyte-webapp/src/pages/ConnectorBuilderPage/LandingPage.tsx

```diff
@@ -0,0 +1,194 @@
+    const { setValues } = useFormikContext<BuilderFormValues>();
+    const [importYamlLoading, setImportYamlLoading] = useState(false);
+
+    const handleYamlUpload = useCallback(
+      async (yaml: string) => {
+        try {
+          let json;
+          try {
+            json = load(yaml) as ConnectorManifest;
+          } catch (e) {
+            if (e instanceof YAMLException) {
+              registerNotification({
+                id: YAML_UPLOAD_ERROR_ID,
+                text: (
+                  <FormattedMessage
+                    id={YAML_UPLOAD_ERROR_ID}
+                    values={{
+                      reason: e.reason,
+                      line: e.mark.line,
+                    }}
+                  />
+                ),
+                type: ToastType.ERROR,
+              });
+            }
+            return;
+          }
+          setYamlIsValid(true);
+
+          let convertedFormValues;
+          try {
+            convertedFormValues = await convertToBuilderFormValues(json, DEFAULT_BUILDER_FORM_VALUES);
+          } catch (e) {
+            switchToYaml();
+            setJsonManifest(json);
+            setShowLandingPage(false);
+            return;
+          }
+
+          switchToUI();
+          setValues(convertedFormValues);
+          setShowLandingPage(false);
+        } finally {
+          if (fileInputRef.current) {
+            fileInputRef.current.value = "";
+          }
+          setImportYamlLoading(false);
+        }
+      },
+      [
+        convertToBuilderFormValues,
+        registerNotification,
+        setJsonManifest,
+        setShowLandingPage,
+        setValues,
+        setYamlIsValid,
+        switchToUI,
+        switchToYaml,
+      ]
+    );
+
+    return (
+      <FlexContainer direction="column" alignItems="center" gap="2xl">
+        <FlexContainer direction="column" gap="md" alignItems="center" className={styles.titleContainer}>
+          <AirbyteLogo />
+          <Heading as="h1" size="lg" className={styles.title}>
+            <FormattedMessage id="connectorBuilder.landingPage.title" />
+          </Heading>
+        </FlexContainer>
+        <Heading as="h1" size="lg">
+          <FormattedMessage id="connectorBuilder.landingPage.prompt" />
+        </Heading>
+        <FlexContainer direction="row" gap="2xl">
+          <input
+            type="file"
+            accept=".yml,.yaml"
+            ref={fileInputRef}
+            style={{ display: "none" }}
+            onChange={(uploadEvent) => {
+              setImportYamlLoading(true);
+              const file = uploadEvent.target.files?.[0];
+              const reader = new FileReader();
+              reader.onload = (readerEvent) => {
+                handleYamlUpload(readerEvent.target?.result as string);
+              };
+              if (file) {
+                reader.readAsText(file);
+              }
+            }}
+          />
```

Comment:
Open to suggestions here, but this seemed like the simplest and most standard way to handle file uploads without using a library. And I couldn't find any libraries that were well supported and lightweight